apps/spline.c

```diff
@@ -32,28 +32,51 @@ static void _apps_spline_paint(apps_spline_t *spline)
 
     path = twin_path_create();
     twin_path_set_cap_style(path, spline->cap_style);
-    twin_path_move(path, spline->points[0].x, spline->points[0].y);
-    twin_path_curve(path, spline->points[1].x, spline->points[1].y,
-                    spline->points[2].x, spline->points[2].y,
-                    spline->points[3].x, spline->points[3].y);
-    twin_paint_stroke(_apps_spline_pixmap(spline), 0xff404040, path,
-                      spline->line_width);
-    twin_path_set_cap_style(path, TwinCapButt);
-    twin_paint_stroke(_apps_spline_pixmap(spline), 0xffffff00, path,
-                      twin_int_to_fixed(2));
-
-    twin_path_empty(path);
-    twin_path_move(path, spline->points[0].x, spline->points[0].y);
-    twin_path_draw(path, spline->points[1].x, spline->points[1].y);
-    twin_paint_stroke(_apps_spline_pixmap(spline), 0xc08000c0, path,
-                      twin_int_to_fixed(2));
-    twin_path_empty(path);
-    twin_path_move(path, spline->points[3].x, spline->points[3].y);
-    twin_path_draw(path, spline->points[2].x, spline->points[2].y);
-    twin_paint_stroke(_apps_spline_pixmap(spline), 0xc08000c0, path,
-                      twin_int_to_fixed(2));
-    twin_path_empty(path);
-    for (i = 0; i < NPT; i++) {
+    if (spline->n_points == 4) {
```

Comment:
Can you generalize the handling for n_points = 3 and 4? That is, share code by avoiding duplication.